plugins/woocommerce/src/Internal/Admin/Settings/PaymentsController.php

```diff
@@ -0,0 +1,209 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\Internal\Admin\Settings;
+
+use Automattic\WooCommerce\Admin\Features\Features;
+use Exception;
+
+defined( 'ABSPATH' ) || exit;
+/**
+ * Payments settings controller class.
+ *
+ * Use this class for hooks and actions related to the Payments settings page.
+ */
+class PaymentsController {
+
+	/**
+	 * The payment service.
+	 *
+	 * @var Payments
+	 */
+	private Payments $payments;
+
+	/**
+	 * Register hooks.
+	 */
+	public function register() {
+		// Because we gate the hooking based on a feature flag,
+		// we need to delay the registration until the 'woocommerce_init' hook.
+		// Otherwise, we end up in an infinite loop.
+		add_action( 'woocommerce_init', array( $this, 'delayed_register' ) );
+	}
+
+	/**
+	 * Delayed hook registration.
+	 */
+	public function delayed_register() {
+		// Don't do anything if the feature is not enabled.
+		if ( ! Features::is_enabled( 'reactify-classic-payments-settings' ) ) {
+			return;
+		}
+
+		add_action( 'admin_menu', array( $this, 'add_menu' ) );
+		add_filter( 'woocommerce_admin_shared_settings', array( $this, 'preload_settings' ) );
+		add_filter( 'woocommerce_admin_allowed_promo_notes', array( $this, 'add_allowed_promo_notes' ) );
+	}
+
+	/**
+	 * Initialize the class instance.
+	 *
+	 * @param Payments $payments The payments service.
+	 *
+	 * @internal
+	 */
+	final public function init( Payments $payments ): void {
+		$this->payments = $payments;
+	}
+
+	/**
+	 * Adds the Payments top-level menu item.
+	 */
+	public function add_menu() {
+		global $menu;
+
+		// The WooPayments plugin must not be active.
+		// When active, WooPayments will own the Payments menu item since it is the native Woo payments solution.
+		if ( $this->is_woopayments_active() ) {
+			return;
+		}
+
+		$menu_title = esc_html__( 'Payments', 'woocommerce' );
+		$menu_icon  = 'data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSI4NTIiIGhlaWdodD0iNjg0Ij48cGF0aCBmaWxsPSIjYTJhYWIyIiBkPSJNODIgODZ2NTEyaDY4NFY4NlptMCA1OThjLTQ4IDAtODQtMzgtODQtODZWODZDLTIgMzggMzQgMCA4MiAwaDY4NGM0OCAwIDg0IDM4IDg0IDg2djUxMmMwIDQ4LTM2IDg2LTg0IDg2em0zODQtNTU2djQ0aDg2djg0SDM4MnY0NGgxMjhjMjQgMCA0MiAxOCA0MiA0MnYxMjhjMCAyNC0xOCA0Mi00MiA0MmgtNDR2NDRoLTg0di00NGgtODZ2LTg0aDE3MHYtNDRIMzM4Yy0yNCAwLTQyLTE4LTQyLTQyVjIxNGMwLTI0IDE4LTQyIDQyLTQyaDQ0di00NHoiLz48L3N2Zz4=';
+		// Link to the Payments settings page.
+		$menu_path = 'admin.php?page=wc-settings&tab=checkout';
+
+		add_menu_page(
+			$menu_title,
+			$menu_title,
+			'manage_woocommerce', // Capability required to see the menu item.
+			$menu_path,
+			null,
+			$menu_icon,
+			56, // Position after WooCommerce Product menu item.
+		);
+
+		// If the store doesn't have any enabled gateways or providers need action, add a notice badge to the Payments menu item.
+		if ( ! $this->store_has_enabled_gateways() || $this->store_has_providers_with_action() ) {
```

Comment:
`$this->store_has_providers_with_action()`

I guess that's a good idea to show the badge if there's any action.